feat: Add 2021Q3 roadmap (#173)
beyondstorage/site: blog/2021-08-06-2021q3-roadmap.md

---
title: "BeyondStorage 2021Q3 Roadmap"
author: "Xuanwo"
author_url: "https://github.com/Xuanwo"
tags: [roadmap]
---

Hello everyone, in this article, we will announce our 2021Q3 roadmap.

## Background

Before starting our plan, let’s revisit our architecture.

![](/img/architecture.svg)

In the past year:

- go-storage API has been more and more stable, with no big break changes anymore.
- Up to 21 services have been or are under development.
- BeyondTP has released several beta versions.

> We have many ideas so far, all of them are listed in [BeyondStorage 2021Q3 Roadmap Draft](https://forum.beyondstorage.io/t/beyondstorage-2021q3-roadmap-draft/164), in this post, we only discuss our roadmap.

> Now we have a solid foundation to support us to go higher and further, we have to think about where should BeyondStorage going on.

Reviewer: ```suggestion
Now we have a solid foundation to support us to go higher and further, we have to think about where BeyondStorage should go.
```

## Roadmap

### More Services Connected

> For now, we have 21 supported services: 9 of them are stable, 1 of them is beta, 10 of them are under development.

Reviewer: 9+1+10 = 20？

Author: ```suggestion
For now, we have 21 supported services: 9 of them are stable, 1 of them is beta, 11 of them are under development.
```

In 2021Q3, we will expand them to 30 services:

> - Already implemented services will become stable.
> - And we will implement 9 more services.

Reviewer: ```suggestion
- Services already implemented will become stable.
- 9 more services will be implemented.
```

After 2021Q3, BeyondStorage will connect almost all major storage services in the world.

### More Releases for BeyondTP

BeyondTP is a neutral data migration service that is built upon go-storage. In the past several months, our development speed is too slow. In 2021Q3, we will accelerate iteration speed.

After 2021Q3, we will release a usable version of BeyondTP.

### More Scene Expansion

> In 2021Q3, we will expand more scenes of BeyondStorage.

Reviewer: ```suggestion
In 2021Q3, we will expand more scenes for BeyondStorage.
```

First of all, we will try to build BeyondFS: a high-performance, POSIX-ish File System based on go-storage. In 2021Q3, we plan to build a first beta release version of BeyondFS.

> Then, we will build an FTP server based on go-storage: BeyondFTP. This service could behave like an FTP gateway of all storage services.

Reviewer: ```suggestion
Then, we will build an FTP server based on go-storage: BeyondFTP. This service could behave like an FTP gateway for all supported storage services.
```
